src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -295,222 +392,266 @@ public void start(final CommandMap command)
-                                getTaggableFilterFrom(pbfRelationConfiguration, configurationMap));
-                    }
-
-                    // Build the appropriate PbfLoader
-                    final PbfLoader loader = new PbfLoader(pbfContext, sparkContext, boundaries,
-                            atlasLoadingOption, codeVersion, dataVersion, task.getAllShards());
-                    final String name = countryName + CountryShard.COUNTRY_SHARD_SEPARATOR
-                            + shard.getName();
-                    final Atlas atlas;
-                    try
-                    {
-                        // Generate the Atlas for this shard
-                        atlas = loader.load(countryName, shard);
-                    }
-                    catch (final Throwable e)
-                    {
-                        throw new CoreException("Building Atlas {} failed!", name, e);
-                    }
+        if (useRawAtlas)
+        {
+            // AtlasLoadingOption isn't serializable, neither is command map. To avoid duplicating
+            // boiler-plate code for creating the AtlasLoadingOption, extract the properties we need
+            // from the command map and pass those around to create the AtlasLoadingOption
+            final Map<String, String> atlasLoadingOptions = extractAtlasLoadingProperties(command);
+
+            // Generate the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countryRawAtlasShardsRDD = getContext()
+                    .parallelize(tasks, tasks.size()).mapToPair(this.generateRawAtlas(boundaries,
+                            sparkContext, atlasLoadingOptions, pbfContext, atlasScheme))
+                    .filter(tuple -> tuple._2() != null);
+
+            // Persist the RDD and save the intermediary state
+            countryRawAtlasShardsRDD.cache();
+            countryRawAtlasShardsRDD.saveAsHadoopFile(
+                    getAlternateSubFolderOutput(output, RAW_ATLAS_FOLDER), Text.class, Atlas.class,
+                    MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE RAW ATLAS **********\n");
+
+            // Slice the raw Atlas and filter any null atlases
+            final JavaPairRDD<String, Atlas> countrySlicedRawAtlasShardsRDD = countryRawAtlasShardsRDD
+                    .mapToPair(this.sliceRawAtlas(boundaries)).filter(tuple -> tuple._2() != null);
+
+            // Persist the RDD and save the intermediary state
+            final String slicedRawAtlasPath = getAlternateSubFolderOutput(output,
+                    SLICED_RAW_ATLAS_FOLDER);
+            countrySlicedRawAtlasShardsRDD.cache();
+            countrySlicedRawAtlasShardsRDD.saveAsHadoopFile(slicedRawAtlasPath, Text.class,
+                    Atlas.class, MultipleAtlasOutputFormat.class, new JobConf(configuration()));
+            logger.info("\n\n********** SAVED THE SLICED RAW ATLAS **********\n");
```

Comment:
Here I believe you can now release the `countryRawAtlasShardsRDD`, because the next step, `countrySlicedRawAtlasShardsRDD` has already been cached.